### PR TITLE
qtstyleplugins: fix gtk2 background

### PR DESCRIPTION
**Pull request by kanashimia**
###### Motivation for this change
gtk2 style is broken, and doesen't render background correctly, but for some reason it works in qt5ct, god knows why.
I found how to fix this on the aur: [link to the patch](https://aur.archlinux.org/cgit/aur.git/tree/0002-fix-gtk2-background.patch?h=qt5-styleplugins)

Maybe this should be backported to stable.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>libsForQt5.qtstyleplugins (libsForQt515.qtstyleplugins ,plasma5Packages.qtstyleplugins)</li>
<li>libsForQt512.qtstyleplugins</li>
<li>libsForQt514.qtstyleplugins</li>
</ul>
</details>

File: pkgs/development/libraries/qtstyleplugins/default.nix [hunk 1]
```diff
@@ -10,7 +10,10 @@ mkDerivation {
     sha256 = "085wyn85nrmzr8nv5zv7fi2kqf8rp1gnd30h72s30j55xvhmxvmy";
   };
 
-  patches = [ ./fix-build-against-Qt-5.15.patch ];
+  patches = [
+    ./fix-build-against-Qt-5.15.patch
+    ./fix-gtk2-background.patch
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qtstyleplugins/default.nix:15]: Please use fetchpatch with ``https://aur.archlinux.org/cgit/aur.git/plain/0002-fix-gtk2-background.patch?h=qt5-styleplugins`` instead.

Comment (kanashimia) [reply pkgs/development/libraries/qtstyleplugins/default.nix:15]: Fetch patch doesen't seem to work, but fetchurl does.

Comment (kanashimia) [reply pkgs/development/libraries/qtstyleplugins/default.nix:15]: Also i don't know, i think this should be consistent, either should provide both patches in the nixpkgs, or download both from the aur.
And i'm not really a fan of relying on the aur package link ;/
What do you think?

Comment (SuperSandro2000) [reply pkgs/development/libraries/qtstyleplugins/default.nix:15]: Wer can't use fetchurl because it does not normalise the patch.

After the download is cached AUR can delete it and we still have it available.

I don't care if both patches are fetched or not but vendoring patches just makes the repo big and they need to be maintained or adapted.

Comment (kanashimia) [reply pkgs/development/libraries/qtstyleplugins/default.nix]: I found out why fetchpatch didn't work - it doesen't prompt for new hash if it is an empty string, but fetchurl does - i will try to see if i can fix it, otherwise i'll file a bug report.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>libsForQt5.qtstyleplugins (libsForQt515.qtstyleplugins ,plasma5Packages.qtstyleplugins)</li>
<li>libsForQt512.qtstyleplugins</li>
<li>libsForQt514.qtstyleplugins</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
libsForQt5.qtstyleplugins:
<details>


warning: unclear-gpl
`lgpl21` is a deprecated license, check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/qtstyleplugins/default.nix:33:5:

```
   |
33 |     license = licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>
